base.rkt

```diff
@@ -140,12 +141,14 @@
     #:description description
     (~optional (~seq #:uses-universal-tagged-syntax? uses-universal-tagged-syntax?))
     (~optional (~seq #:analyzers analyzers))
+    (~optional (~seq #:suggested-fixes suggested-fixes))
```

Comment:
Thinking about this more, with this API there's no way to make the replacement template optional and statically enforce that an optional replacement template requires that the value `'none` be passed here.

I think this should work differently. Instead of a `#:suggested-fixes expr` clause at the beginning of the rule declaration, the `replacement` template at the end of the declaration should be allowed to instead be the keyword `#:no-suggestion`. Like this:

```scheme
(define-refactoring-rule equal-against-self-warning
  #:description "Comparing a value to itself will always return true."
  #:literals (equal?)
  (equal? a:id a2:id)
  #:when (free-identifier=? (attribute a) (attribute b))
  #:no-suggestion)
```

(And similarly for `define-definition-context-refactoring-rule`.)